### PR TITLE
Feature/async incremental recorder loading

### PR DESCRIPTION
**Pull request by RastaChaum**
This pull request introduces a new framework for incremental, asynchronous extraction of historical heating cycle data, with the goal of improving performance, responsiveness, and scalability during large data loads in Home Assistant. The update includes new value objects and services for managing extraction tasks, a queue-based orchestrator for daily extraction, and supporting utilities for calculating extraction date ranges. Several interfaces for on-demand and cancellable extraction are stubbed out for future implementation.

The most important changes are:

**Incremental Extraction Framework**

- Added new async interface methods to `HeatingCycleLifecycleManager` for triggering, cancelling, and checking the status of incremental and on-demand extractions, including callbacks for progressive cache updates. These methods are currently stubbed with `NotImplementedError` for future implementation.

**Extraction Task Value Objects**

- Introduced `RecordingExtractionTask` and `ExtractionTaskState` value objects in `recording_extraction_task.py`, encapsulating the state and metadata for single-day extraction tasks, and made them available in the domain's `__init__.py`. [[1]](diffhunk://#diff-4684d9ebf8eb653a13bd2af0ac99c2c9a968201b801c8d5098936c9192c8f878R1-R55) [[2]](diffhunk://#diff-2cde830faa8b33680aaad114ff0b3688ca0d4cc7b45e06b762e1aa9e67ed50e2R13) [[3]](diffhunk://#diff-2cde830faa8b33680aaad114ff0b3688ca0d4cc7b45e06b762e1aa9e67ed50e2R30-R31)

**Extraction Queue Orchestration**

- Implemented `RecordingExtractionQueue`, a queue-based, asynchronous orchestrator for sequential daily extraction from the Home Assistant Recorder, supporting progressive cache population, cancellation, and progress tracking.

**Extraction Date Range Calculation**

- Added `ExtractionDateRangeCalculator`, a utility service to calculate optimal extraction date ranges based on retention policy, cache state, and current date, supporting both full and refresh extraction scenarios.

**Minor Bugfix**

- Fixed the logic for determining the `current_hour` in `lhs_lifecycle_manager.py` to correctly handle cases where `dt_util` is `None`.## Description

## Related Issue

Fixes #(issue number)

## Type of Change

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [ ] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 📝 Documentation update
- [ ] 🔧 Refactoring (no functional changes)
- [ ] ✅ Test coverage improvement
- [ ] 🏗️ Infrastructure/build changes

## Changes Made

- 
- 
- 

## Testing Performed

- [ ] All existing unit tests pass
- [ ] Added new unit tests for changes
- [ ] Tested manually in Home Assistant
- [ ] Tested with multiple VTherm configurations
- [ ] Tested edge cases

### Test Details

**Environment:**
- Home Assistant Version: 
- Python Version: 
- VTherm Version: 

**Test Scenarios:**
1. 
2. 
3. 

## Checklist

- [ ] My code follows the project's style guidelines (PEP 8, Black formatting)
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings or errors
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published
- [ ] I have updated CHANGELOG.md in the `[Unreleased]` section
- [ ] Domain layer has NO `homeassistant.*` imports (if applicable)
- [ ] All external interactions use ABCs/interfaces (if applicable)
- [ ] Value objects are immutable with `@dataclass(frozen=True)` (if applicable)

## Architecture Compliance (for code changes)

- [ ] Domain layer remains independent of Home Assistant
- [ ] Business logic is in domain services, not infrastructure
- [ ] All infrastructure changes implement existing interfaces
- [ ] No business rules in adapters
- [ ] Tests use mocks for external dependencies

## Breaking Changes

**Impact:**
- 

**Migration Guide:**
1. 
2. 
3. 

## Reviewer Checklist

- [ ] Code follows project architecture (DDD principles)
- [ ] Tests are comprehensive and pass
- [ ] Documentation is clear and complete
- [ ] Breaking changes are clearly documented
- [ ] CHANGELOG.md is updated
- [ ] Commit messages follow Conventional Commits

### CODE REVIEW

File: custom_components/intelligent_heating_pilot/infrastructure/adapters/recording_extraction_queue.py [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+        )
+
+        if self._heating_cycle_service is None:
+            raise RuntimeError("HeatingCycleService is required for extraction")
+
+        try:
+            # Create time window for this day with UTC timezone
+            start_time = datetime.combine(extraction_date, time.min).replace(tzinfo=timezone.utc)
+            end_time = datetime.combine(extraction_date, time.max).replace(tzinfo=timezone.utc)
+
```
[the remark this answers is not in the document]
Comment (RastaChaum) [reply custom_components/intelligent_heating_pilot/infrastructure/adapters/recording_extraction_queue.py]: @Copilot: implement fix

**Review by RastaChaum — COMMENTED** [review]
Vérifie que toutes les extractions de données du Recorder HA se font : 
- par une tache asynchrone
- par tranche de 24H
- qu'une tache d'extraction finie entraine une mise à jour du cache de manière synchrone.

File: custom_components/intelligent_heating_pilot/application/heating_cycle_lifecycle_manager.py
Comment (RastaChaum) [inline custom_components/intelligent_heating_pilot/application/heating_cycle_lifecycle_manager.py]: La logique complète de `HeatingCycleLifecycleManager` doit être revue. 

1. A chaque évènement (`startup`, `on_retention_change`, `on_24h_timer`), le manager doit :

- calculer sa fenêtre d'extraction des données
- vérifier si il possède déjà en cache des cycles pour `start_time` et `end_time`:  
  - si c'est le cas -> cache à jour, aucune extraction à faire
  - sinon, il regarde le nombre de jour manquant entre `start_time` et son plus vieux cycle, et entre `end_time` et son cycle le plus récent. Et il va alors extraire les journées manquantes avec des tâches asynchrone, par step de 24H...

- mettre à jour son cache le cas échéant

2. La fin d'une tâche asychrone entraîne 
- la mise à jour du cache de façon synchrone. 
- le recalcule des LHS (Contextual & Globa) et des anticipation_data (si cache mis à jour)


Il ne doit plus exister d'extraction de données du Recorder synchrone !

NB: ajouter également un test pour que la `end_time` d'extraction ne soit pas la date du jour, mais la veille au plus tard -> éviter les extractions partiels de cycles pour la journée courante.

**Review by RastaChaum — COMMENTED** [review]
@copilot : prend en compte les remarques
